src/attachScopes.ts

```diff
@@ -100,7 +100,7 @@ const attachScopes: AttachScopes = function attachScopes(ast, propertyName = 'sc
 			if (node.type === 'CatchClause') {
 				newScope = new Scope({
 					parent: scope,
-					params: [node.param],
+					params: node.param ? [node.param] : undefined,
```

Comment:
Should this not be the empty array? Just seems like it would need a lot of guards for the `params: undefined` case, but if that is already handled then that is ok.

Reply:
@guybedford I have no idea. This prevented the code from throwing `TypeError: Cannot read property 'type' of null` when I changed it. 

I was following the example above for `BlockStatement` which does not assign `params`.

If you think an empty array is better, that seems to work too.

Reply:
By the current implementation, it does not really matter what you do, but again I would love to see a test to make sure this keeps working in the future.

Reply:
@lukastaegert Thanks, I added a test 👍

Reply:
Can we make this `null` instead of `undefined` to match the ESTree approach then?

Reply:
@guybedford It can't be null due to the interface:

```
interface ScopeOptions {
	parent?: AttachedScope;
	block?: boolean;
	params?: Array<Node>;
}
```

It can only be `undefined` or `Array`